backend/main_ad_manager.py

```diff
@@ -21,21 +16,21 @@ def find_ad_for_subscriber(subscriber: Subscriber, ads: List[Ad]) -> Optional[Ad
     return None
 
 def send_ad(data, context) -> str:
-    logging.info("********* SENDING ADS... *********")
+    custom_logger.info("********* SENDING ADS... *********")
```

Comment:
I think a more proper way of doing this is by using string formatting, e.g.:
```python
custom_logger.info(f'{" SENDING ADS... ":*^n}')
```
where the asterisk signifies what the string should be padded with, and `n` is the number of bytes the line should print.

The script above would be printed as `******* SENDING ADS... *******` with `n=30`.

If padding like that is used, it might be better to just implement it right into the `custom_logger`, so you don't need to pad by yourself.